include/soci/soci-platform.h

```diff
@@ -37,6 +37,9 @@
 //base class must have dll interface
 #pragma warning(disable:4251 4275)
 
+// As long as we don't require C++17, we must disable the warning
+// "conditional expression is constant"
```

Comment:
Could you please add the context here, i.e. where would this warning be given?

Reply:
TBH I don't understand this error either. But as it didn't seem to be local to `type-holder.h`, I suppressed it here.
The failing build was this one: https://ci.appveyor.com/project/SOCI/soci/builds/49285927/job/fikvgmaiqq11a9np

Reply:
I guess the line refers to the `val > static_cast<U>(t_max)` test? If so, I can understand why the compiler would warn about it, as it could be always false for some types, but I'm not sure how to avoid it. Using pragma push/pop around this test might be better than suppressing the warning globally, however, it's a potentially useful one.

Reply:
Ah, I see. I'll move the pragma push/pop to this section.